common/flatpak-transaction.c

```diff
@@ -2431,6 +2444,82 @@ resolve_all_ops (FlatpakTransaction *self,
   return TRUE;
 }
 
+
+static gboolean
+request_tokens_for_remote (FlatpakTransaction *self,
+                           const char         *remote,
+                           GList              *ops,
+                           GCancellable       *cancellable,
+                           GError            **error)
+{
+  g_autoptr(GString) refs_as_str = g_string_new ("");
+  GList *l;
+  const char *token;
+
+  for (l = ops; l != NULL; l = l->next)
+    {
+      FlatpakTransactionOperation *op = l->data;
+
+      if (l != ops)
+        g_string_append (refs_as_str, ", ");
+      g_string_append (refs_as_str, op->ref);
+    }
+
+  g_debug ("Requesting tokens for remote %s, refs: %s", remote, refs_as_str->str);
+
+  token = g_getenv ("FLATPAK_TEST_TOKEN");
+  if (token == NULL)
+    token = "default-token";
+
+  g_debug ("Got token %s", token);
+
+  for (l = ops; l != NULL; l = l->next)
+    {
+      FlatpakTransactionOperation *op = l->data;
+
+      op->resolved_token = g_strdup (token);
```

Comment:
Paranoia: I’d add a `g_assert (op->resolved_token == NULL)` or `g_clear_pointer (&op->resolved_token, g_free)` first, just in case this ends up getting called twice.